src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,221 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        BgIkanaDharma_BeginCutsceneState(this);
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
+
+void BgIkanaDharma_BeginCutsceneState(BgIkanaDharma* this) {
+    this->cutsceneFramesRemaining = 0xA;
+    this->actionFunc = BgIkanaDharma_UpdateCutscene;
+}
+
+void BgIkanaDharma_UpdateCutscene(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    if (this->cutsceneFramesRemaining > 0) {
+        this->cutsceneFramesRemaining--;
+        if (this->cutsceneFramesRemaining == 0) {
+            if (sFirstHitBgIkanaDharma == this) {
+                sFirstHitBgIkanaDharma = NULL;
+            }
+            ActorCutscene_Stop(this->dyna.actor.cutscene);
+        }
+    }
+    if (Math_StepToF(&this->dyna.actor.scale.y, 0.0f, 1.0f / 300.0f) != 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        this->dyna.actor.scale.x = this->dyna.actor.scale.y * 3.0f;
+        this->dyna.actor.scale.z = this->dyna.actor.scale.y * 3.0f;
+        BgIkanaDharma_CreateParticles(this, globalCtx);
+    }
+}
+
+void BgIkanaDharma_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkanaDharma* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc == BgIkanaDharma_UpdateNormalState) {
+        DynaPolyActor* actorBelow;
+        u32 pad, pad2;
+        func_800B4AEC(globalCtx, &this->dyna.actor, 30.0f);
+        actorBelow = DynaPoly_GetActor(&globalCtx->colCtx, this->dyna.actor.floorBgId);
+        if (actorBelow == NULL) {
+            Actor_MoveWithGravity(&this->dyna.actor);
+            Actor_UpdateBgCheckInfo(globalCtx, &this->dyna.actor, 0.0f, 0.0f, 0.0f, 4);
+            if (this->dyna.actor.bgCheckFlags & 2) {
+                s16 quake = Quake_Add(globalCtx->cameraPtrs[globalCtx->activeCamera], 3);
+                Quake_SetSpeed(quake, 21536);
+                Quake_SetQuakeValues(quake, 4, 0, 0, 0);
+                Quake_SetCountdown(quake, 12);
+                Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_BLOCK_BOUND);
+            }
+        } else {
+            if (actorBelow->actor.id == ACTOR_BG_IKANA_DHARMA) {
+                this->dyna.actor.world.pos.y = actorBelow->actor.world.pos.y + 60.0f;
+            } else {
+                this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+            }
+            this->dyna.actor.velocity.y = 0.0f;
+        }
+    } else {
+        f32 phi_f0 = this->dyna.actor.scale.x * 300.0f;
+        if (phi_f0 < 2.0f) {
+            phi_f0 = 2.0f;
+        }
+        Actor_MoveWithGravity(&this->dyna.actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->dyna.actor, 20.0f, phi_f0, 0.0f, 5);
+    }
+    Actor_SetFocus(&this->dyna.actor, 40.0f);
+}
+
+void BgIkanaDharma_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkanaDharma* this = THIS;
+    Gfx_DrawDListOpa(globalCtx, object_ikana_obj_DL_0008C8);
```

Comment:
Can document the object too if you want, this one's likely to be very simple. There's a section in the tutorial about it.